src/main/java/subway/adapter/out/graph/ShortPathAdapter.java

```diff
@@ -0,0 +1,55 @@
+package subway.adapter.out.graph;
+
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.WeightedMultigraph;
+import subway.adapter.out.graph.dto.RouteDto;
+import subway.application.port.out.graph.ShortPathPort;
+import subway.domain.Sections;
+import subway.domain.Station;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ShortPathAdapter implements ShortPathPort {
+
+    @Override
+    public RouteDto findSortPath(final Station fromStation, final Station toStation, final Map<Long, Sections> sectionsByLine) {
+        WeightedMultigraph<Station, LineWeightedEdge> graph
+                = new WeightedMultigraph<>(LineWeightedEdge.class);
+        for (Long lineId : sectionsByLine.keySet()) {
+            sectionsByLine.get(lineId).getSortedStations()
+                    .forEach(graph::addVertex);
+
+            sectionsByLine.get(lineId).getSections().forEach(section -> {
+                LineWeightedEdge edge = new LineWeightedEdge(section.getLineId());
+                graph.addEdge(section.getUpStation(), section.getDownStation(), edge);
+                graph.setEdgeWeight(edge, section.getDistance());
+            });
+        }
+
+        DijkstraShortestPath dijkstraShortestPath
+                = new DijkstraShortestPath(graph);
```

Comment:
제네릭 사용시 raw type을 사용하지 말라는 권고사항이 있는데요([참고](https://rules.sonarsource.com/java/RSPEC-3740))
해당 코드를 아래처럼 변경해보면 어떨까요?
```suggestion
        DijkstraShortestPath<Station, LineWeightedEdge> dijkstraShortestPath
                = new DijkstraShortestPath<>(graph);
```

Reply:
아 제네릭타입이였군요(몰랐습니다..),,,🥹 수정하겠습니다!!!!!